src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,312 @@
+    private final Map<String, Object> fieldMap;
+
+    private final ClusterService clusterService;
+
+    private final IndicesService indicesService;
+
+    private final AnalysisRegistry analysisRegistry;
+
+    private final Environment environment;
+
+    public DocumentChunkingProcessor(
+        String tag,
+        String description,
+        Map<String, Object> fieldMap,
+        Map<String, Object> algorithmMap,
+        Environment environment,
+        ClusterService clusterService,
+        IndicesService indicesService,
+        AnalysisRegistry analysisRegistry
+    ) {
+        super(tag, description);
+        this.fieldMap = fieldMap;
+        this.environment = environment;
+        this.clusterService = clusterService;
+        this.indicesService = indicesService;
+        this.analysisRegistry = analysisRegistry;
+        validateAndParseAlgorithmMap(algorithmMap);
+    }
+
+    public String getType() {
+        return TYPE;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void validateAndParseAlgorithmMap(Map<String, Object> algorithmMap) {
+        if (algorithmMap.size() != 1) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as [" + ALGORITHM_FIELD + "] must contain and only contain 1 algorithm"
+            );
+        }
+        Entry<String, Object> algorithmEntry = algorithmMap.entrySet().iterator().next();
+        String algorithmKey = algorithmEntry.getKey();
+        Object algorithmValue = algorithmEntry.getValue();
+        Set<String> supportedChunkers = ChunkerFactory.getAllChunkers();
+        if (!supportedChunkers.contains(algorithmKey)) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as chunker algorithm ["
+                    + algorithmKey
+                    + "] is not supported. Supported chunkers types are "
+                    + supportedChunkers
+            );
+        }
+        if (!(algorithmValue instanceof Map)) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as [" + algorithmKey + "] parameters cannot be cast to [" + Map.class.getName() + "]"
+            );
+        }
+        Map<String, Object> chunkerParameters = (Map<String, Object>) algorithmValue;
+        if (Objects.equals(algorithmKey, FIXED_TOKEN_LENGTH_ALGORITHM)) {
+            chunkerParameters.put(FixedTokenLengthChunker.ANALYSIS_REGISTRY_FIELD, analysisRegistry);
+        }
+        this.chunker = ChunkerFactory.create(algorithmKey, chunkerParameters);
+        if (chunkerParameters.containsKey(MAX_CHUNK_LIMIT_FIELD)) {
+            String maxChunkLimitString = chunkerParameters.get(MAX_CHUNK_LIMIT_FIELD).toString();
+            if (!(NumberUtils.isParsable(maxChunkLimitString))) {
+                throw new IllegalArgumentException(
+                    "Parameter [" + MAX_CHUNK_LIMIT_FIELD + "] cannot be cast to [" + Number.class.getName() + "]"
```

Comment:
same here, please check other message strings as well, there are many in other classes touched by this PR

Reply:
Already changed.